### PR TITLE
Fix shards regression

### LINKED ISSUE #486: Shards regression in 0.18
It looks like there was a shards regression introduced in v0.18. MWE:
```julia
julia> Clang.JLLEnvs.get_environment_info("armv7l-linux-gnueabihf")
ERROR: KeyError: key "GCCBootstrap-armv7l-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked" not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:498 [inlined]
 [2] get_environment_info(p::Base.BinaryPlatforms.Platform, version::VersionNumber)
   @ Clang.JLLEnvs ~/Documents/code/Clang.jl/src/platform/env.jl:42
 [3] get_environment_info
   @ ~/Documents/code/Clang.jl/src/platform/env.jl:46 [inlined]
 [4] get_environment_info(triple::String)
   @ Clang.JLLEnvs ~/Documents/code/Clang.jl/src/platform/env.jl:46
 [5] top-level scope
   @ REPL[8]:1
```

but on v0.17:
```julia
julia> Clang.JLLEnvs.get_environment_info("armv7l-linux-gnueabihf")
2-element Vector{@NamedTuple{id::String, url::String, chk::String}}:
 (id = "8829cc041d600db0e8623c301d4ae8906fa1f060", url = "https://github.com/JuliaPackaging/Yggdrasil/releases/download/GCCBootstrap-v4.8.5+3/GCCBootstrap-armv7l-linux-gnueabihf.v4.8.5.x86_64-linux-musl.unpacked.tar.gz", chk = "2e4b2d29bd1b608ad54d664f16b6d35003f1f29eca3242ccdcc096baf186b9b0")
 (id = "347b47a165098e5e8cbbe03f518757629137c006", url = "https://github.com/JuliaPackaging/Yggdrasil/releases/download/PlatformSupport-v2021.8.10/PlatformSupport-armv7l-linux-gnueabihf.v2021.8.10.x86_64-linux-musl.unpacked.tar.gz", chk = "093d3930e2b3c10775cc854fdda49aabfdd22afb60cb50dc08293a283dbefaf0")
```

Maybe a good test to add (based on my use case) is :
```julia
Clang.get_default_args.(Clang.JLLEnvs.JLL_ENV_TRIPLES)
```
just to make sure that doesn't error

### PR DESCRIPTION
**Pull request by Octogonapus**
This PR is my attempt at fixing the shards regression in #486. I added a test that checks all the default triples.

Fixes #486.

### CODE REVIEW

File: src/platform/target.jl [hunk 1]
```diff
@@ -11,23 +11,25 @@ const JLL_ENV_CLANG_TARGETS_MAPPING = Dict(
...
+    "x86_64-unknown-freebsd13.2"=>"x86_64-unknown-freebsd13.2",
     "x86_64-w64-mingw32"=>"x86_64-w64-windows-gnu",
 )
 
 triple2target(triple::String) = get(JLL_ENV_CLANG_TARGETS_MAPPING, triple, "unknown")
 
 function __triplet(p::Platform)
     for k in keys(p.tags)
-        if k != "arch" && k != "os" && k != "libc"
+        if k != "arch" && k != "os" && k != "libc" && k != "call_abi" && k != "os_version"
```
Comment (Octogonapus) [inline src/platform/target.jl:22]: `call_abi` is needed for armv7l, `os_version` is needed for freebsd and sometimes darwin

File: src/platform/target.jl [hunk 2]
```diff
@@ -11,23 +11,25 @@ const JLL_ENV_CLANG_TARGETS_MAPPING = Dict(
...
             delete!(p.tags, k)
         end
     end
     t = triplet(p)
-    if os(p) == "macos" && arch(p) == "x86_64"
-        t *= "14"
-    elseif os(p) == "macos" && arch(p) == "aarch64"
-        t *= "20"
+    if os_version(p) === nothing
```
Comment (Octogonapus) [inline src/platform/target.jl:27]: I don't know why the user wouldn't specify the version, but here I am just maintaining backward compatibility if they didn't.

**Comment by Gnimuc** [comment]
looks like there are download issues on windows platforms.

File: test/jllenvs.jl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+@testset "get_default_args for all triples" begin
+    # just testing this does not crash on any default triple
+    Clang.get_default_args.(Clang.JLLEnvs.JLL_ENV_TRIPLES)
+end
```
Comment (Gnimuc) [inline test/jllenvs.jl:4]: this test downloads artifacts for all platforms. running it every time is a bit overkill. 

is there a way to somehow cache it?

Comment (Gnimuc) [reply test/jllenvs.jl:4]: or maybe we can control it by env variables.

Comment (Octogonapus) [reply test/jllenvs.jl:17]: sorry i didn't realize it did that, i've simplified the test

**Comment by Octogonapus** [comment]
> looks like there are download issues on windows platforms.

I'm not sure what that error means, do you know?

**Comment by Gnimuc** [comment]
> > looks like there are download issues on windows platforms.
> 
> I'm not sure what that error means, do you know?

no. probably just a mysterious connection issue.

**Comment by Gnimuc** [comment]
The new test looks good. Thanks!

**Comment by Gnimuc** [comment]
just noticed we just hit 1000 commits ;)

**Pull request merged**